sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -217,9 +222,14 @@ private[sql] case class DataSourceScan(
     // TODO: The abstractions between this class and SqlNewHadoopRDD makes it difficult to know
     // here which path to use. Fix this.
 
-    ctx.INPUT_ROW = row
     ctx.currentVars = null
-    val columns1 = exprs.map(_.gen(ctx))
+    val colVars = output.zipWithIndex.map(x => ctx.freshName("col" + x._2))
+    val colDeclarations = colVars.zipWithIndex.map(x => {
```

Comment:
it's better to written as this:
colVars.zipWithIndex.map { case (name, i) =>
   ....
}